src/txn/txn_recover.c

```diff
@@ -503,6 +504,23 @@ __recovery_set_ckpt_base_write_gen(WT_RECOVERY *r)
     return (ret);
 }
 
+/*
+ * __recovery_txn_setup_initial_state --
+ *     Setup the transaction initial state required by rollback to stable.
+ */
+static int
+__recovery_txn_setup_initial_state(WT_SESSION_IMPL *session, WT_RECOVERY *r)
+{
+    WT_RET(__recovery_set_checkpoint_timestamp(r));
+    WT_RET(__recovery_set_oldest_timestamp(r));
+    WT_RET(__recovery_set_checkpoint_snapshot(session));
+
+    /* Update the global time window state to have consistent view from global visibility rules */
```

Comment:
Can you update this comment to indicate that order is important here? Something like: "Now that timestamps extracted from the checkpoint metadata have been configured, configure the pinned timestamp."

Reply:
Thanks, fixed it as suggested.